meson.build

```diff
@@ -57,88 +70,73 @@ else
   conf.set('LIBXMLPP_VERSION_2_6', 0)
 else
-  libxmlpp = dependency('libxml++-2.6')
+  libxmlpp = dependency('libxml++-2.6', required: true)
   conf.set('LIBXMLPP_VERSION_2_6', 1)
   conf.set('LIBXMLPP_VERSION_3_0', 0)
+  warning('libxml++-2.6 is deprecated.  Use a more recent version if possible.')
 endif
 
-#OpenCV may be referred to as opencv or opencv4
+# OpenCV may be referred to as opencv or opencv4
+opencv = nulldependency
 if get_option('usewebcam')
-  message('OpenCV build requested')
   opencv = dependency('opencv', required: false)
-  if opencv.found()
-      add_project_arguments('-DUSE_OPENCV', language : 'cpp')
-  else
-    opencv = dependency('opencv4', method : 'pkg-config', required: false)
-    if opencv.found()
-      add_project_arguments('-DUSE_OPENCV', language : 'cpp')
-    else
-      error('OpenCV not found but webcam build requested')
-    endif
+  if not opencv.found()
+    opencv = dependency('opencv4', required: true)
   endif
-else
-  opencv=dependency('',required: false)
 endif
 
-#Portmidi is maybe not found in pkg-config, use find library as an alternative
+# Portmidi is maybe not found in pkg-config, use find library as an alternative,
+# with both names libportmidi (on ubuntu) and portmidi (on OSX)
+portmidi = nulldependency
 if get_option('usemididrum')
-  message('Portmidi build requested')
   portmidi = dependency('portmidi', required: false)
-  if portmidi.found()
-    add_project_arguments('-DUSE_PORTMIDI', language: 'cpp')
-  else
+  if not portmidi.found()
     portmidi = compiler.find_library('libportmidi', required: false)
-    if portmidi.found()
-      add_project_arguments('-DUSE_PORTMIDI', language: 'cpp')
-    else
-      error('PORTMIDI not found but MIDI drum support build requested')
+    if not portmidi.found()
+      portmidi = compiler.find_library('portmidi', required: true)
     endif
   endif
-else
-  portmidi=dependency('',required: false)
 endif
 
-#Cpprest may not be in pkg-config
-#Also, libssl and libcrypto are only needed when webserver support is requested
+# Note: cpprest may not be in pkg-config
+cpprest = nulldependency
+libssl = nulldependency
+libcrypto = nulldependency
 if get_option('usewebserver')
-  message('Webserver build requested')
-  cpprest=dependency('cpprest', required : false)
-  if cpprest.found()
-    add_project_arguments('-DUSE_WEBSERVER', language: 'cpp')
-  else
-    cpprest=compiler.find_library('libcpprest', required:false)
-    if cpprest.found()
-      add_project_arguments('-DUSE_WEBSERVER', language: 'cpp')
-    else
-      error('CppRestSdk not found but webserver build requested')
+  libssl = dependency('libssl', required: true)
+  libcrypto = dependency('libcrypto', required: true)
+  cpprest = dependency('cpprest', required: false)
+  if not cpprest.found()
+    cpprest = compiler.find_library('libcpprest', required: false)
+    if not cpprest.found()
+      cpprest = compiler.find_library('cpprest', required: true)
```

Comment:
Please add a case for finding cpprest using CMake, similar to what I outlined in the discussion.

Reply:
> > ```diff
> > cpprest=dependency('cpprestsdk', required: false, method: 'cmake', modules: 'cpprestsdk::cpprest')
> > ```
> 
> It is strange that it would work with specifying method: 'cmake' explicitly as the default detection logic is
> 
> 1. pkg-config
> 2. cmake
> 3. extraframework
> 
> Could you try with this PR script and removing the method: 'cmake' parameter ?

It doesn't work because the CMake package is `cpprestsdk::cpprest`, not `cpprest`, I also found it odd at first.

Reply:
Done.